src/merkle/store.rs

```diff
@@ -0,0 +1,452 @@
+        Ok(roots.iter().next().unwrap().into())
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the node at `index` rooted on the tree `root`.
+    ///
+    /// # Errors
+    ///
+    /// This will return `NodeNotInStorage` if the element is not present in the store.
+    pub fn get_node(&self, root: Word, index: NodeIndex) -> Result<Word, MerkleError> {
+        let mut hash: RpoDigest = root.into();
+        for bit in index.bit_iterator().rev() {
+            let node = self
+                .nodes
+                .get(&hash)
+                .ok_or(MerkleError::NodeNotInStorage(hash.into(), index))?;
+            hash = if bit { node.right } else { node.left }
+        }
+
+        Ok(hash.into())
+    }
+
+    /// Returns the path for the node at `index` rooted on the tree `root`.
+    ///
+    /// # Errors
+    ///
+    /// This will return `NodeNotInStorage` if the element is not present in the store.
+    pub fn get_path(
+        &self,
+        root: Word,
+        index: NodeIndex,
+    ) -> Result<(Word, MerklePath), MerkleError> {
+        let mut hash: RpoDigest = root.into();
+        let mut path = Vec::new();
+        let node = RpoDigest::default();
+        for bit in index.bit_iterator() {
+            let node = self
+                .nodes
+                .get(&hash)
+                .ok_or(MerkleError::NodeNotInStorage(hash.into(), index))?;
+
+            hash = if bit {
+                path.push(node.left.into());
+                node.right
+            } else {
+                path.push(node.right.into());
+                node.left
+            }
+        }
+
+        Ok((node.into(), MerklePath::new(path)))
+    }
+
+    // DATA MUTATORS
+    // --------------------------------------------------------------------------------------------
+
+    pub fn set_node(
+        &mut self,
+        root: Word,
+        index: NodeIndex,
+        value: Word,
+    ) -> Result<Word, MerkleError> {
+        let (current_node, path) = self.get_path(root, index)?;
+        if current_node != value {
+            self.add_merkle_path(index.value(), value, path)
+        } else {
+            Ok(root)
+        }
+    }
+
+    pub fn merge_roots(&mut self, root1: Word, root2: Word) -> Result<Word, MerkleError> {
+        if !self.nodes.contains_key(&root1.into()) {
+            Err(MerkleError::NodeNotInStorage(root1, NodeIndex::new(0, 0)))
+        } else if !self.nodes.contains_key(&root1.into()) {
+            Err(MerkleError::NodeNotInStorage(root2, NodeIndex::new(0, 0)))
+        } else {
+            let parent: Word = Rpo256::merge(&[root1.into(), root2.into()]).into();
+            self.nodes.insert(
+                parent.into(),
+                Node {
+                    left: root1.into(),
+                    right: root2.into(),
+                },
+            );
+
+            Ok(parent)
+        }
+    }
+}
```

Comment:
If we convert `root1` and `root2` to `RpoDigest` upfront, we can avoid most `into()` calls. Fo example, something like:
```Rust
let root1: RpoDigest = root1.into();
let root2: RpoDigest = root2.into();
```